### PR TITLE
SUPP: Raise exception on backends where rows_with_max_lookback can't be implemented

### LINKED ISSUE #1833: SUPP: Raise exception on backends where rows_with_max_lookback can't be implemented
Follow up from https://github.com/ibis-project/ibis/pull/1825

### PR DESCRIPTION
**Pull request by emilyreff7**
Closes #1833

### CODE REVIEW

File: ibis/impala/tests/test_window.py [hunk 1]
```diff
@@ -129,6 +123,15 @@ def test_window_frame_specs(con, window, frame):
     assert_sql_equal(expr, expected)
 
 
+def test_window_rows_with_max_lookback(con):
+    with pytest.raises(NotImplementedError):
```
Comment (cpcloud) [inline ibis/impala/tests/test_window.py]: Can you put all of the code that doesn't raise an exception outside the `with` block? That way if something else raises that error spuriously we don't swallow it and think that that code is working.

File: ibis/pandas/execution/tests/test_window.py [hunk 2]
```diff
@@ -296,24 +296,14 @@ def test_batting_rolling(batting, batting_df, sort_kind):
 
 
 def test_batting_rolling_with_mlb(batting, batting_df, sort_kind):
-    rows_with_mlb = rows_with_max_lookback(5, ibis.interval(days=10))
-    expr = batting.mutate(
-        more_values=lambda t: t.G.sum().over(
-            ibis.trailing_window(rows_with_mlb, order_by=t.yearID)
+    with pytest.raises(NotImplementedError):
```
Comment (cpcloud) [inline ibis/pandas/execution/tests/test_window.py]: Same comment as above.

File: ibis/pandas/execution/window.py [hunk 3]
```diff
@@ -90,6 +90,10 @@ def execute_window_op(
         **kwargs,
     )
 
+    if window.max_lookback is not None:
+        raise NotImplementedError('Rows with max lookback is not implemented '
+                                  'for pandas backends.')
```
Comment (cpcloud) [inline ibis/pandas/execution/window.py]: I think you can use singular `backend` here

File: ibis/sql/postgres/tests/test_functions.py [hunk 4]
```diff
@@ -886,6 +887,16 @@ def test_rolling_window(alltypes, func, df):
     tm.assert_series_equal(result, expected)
 
 
+def test_rolling_window_with_mlb(alltypes):
+    with pytest.raises(NotImplementedError):
```
Comment (cpcloud) [inline ibis/sql/postgres/tests/test_functions.py]: Same comment re putting code outside the with.

File: ibis/impala/compiler.py [hunk 5]
```diff
@@ -334,6 +334,10 @@ def _window(translator, expr):
 def _format_window(translator, op, window):
     components = []
 
+    if window.max_lookback is not None:
+        raise NotImplementedError('Rows with max lookback is not implemented '
+                                  'for impala backends.')
```
Comment (cpcloud) [inline ibis/impala/compiler.py]: Let's make this `Impala-based backends`.

File: ibis/sql/alchemy.py [hunk 6]
```diff
@@ -564,6 +564,10 @@ def _window(t, expr):
         arg = _cumulative_to_window(t, arg, window)
         return t.translate(arg)
 
+    if window.max_lookback is not None:
+        raise NotImplementedError('Rows with max lookback is not implemented '
+                                  'for sqlalchemy backends.')
```
Comment (cpcloud) [inline ibis/sql/alchemy.py]: Let's make this `SQLAlchemy-based backends`

File: ibis/impala/tests/test_window.py (hunk 1 above)
Comment (emilyreff7) [reply ibis/impala/tests/test_window.py]: Yes, fixed in next commit

File: ibis/pandas/execution/window.py (hunk 3 above)
Comment (emilyreff7) [reply ibis/pandas/execution/window.py]: Fixed in next commit

File: ibis/impala/compiler.py (hunk 5 above)
Comment (emilyreff7) [reply ibis/impala/compiler.py]: Fixed in next commit

File: ibis/pandas/execution/tests/test_window.py (hunk 2 above)
Comment (emilyreff7) [reply ibis/pandas/execution/tests/test_window.py]: Fixed in next commit

File: ibis/sql/postgres/tests/test_functions.py (hunk 4 above)
Comment (emilyreff7) [reply ibis/sql/postgres/tests/test_functions.py]: Fixed in next commit

File: ibis/sql/alchemy.py (hunk 6 above)
Comment (emilyreff7) [reply ibis/sql/alchemy.py]: Fixed in next commit

**Pull request merged**